### ISSUE TITLE
Bug: [no-array-constructor] Does not work when type is provided

### ISSUE THREAD

**Issue by DerZyklop** [typescript-eslint/typescript-eslint#7105]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.0.4&sourceType=module&code=PTAEFMGcBsEsDsAuBaAJrSBDARtcz5wAPFOQ0AAUQE8AHKAYwCdZaUozFh4B7ZTJk0zVkDHvEiImAVwaIeTAFBiJiUEyjToiAEygAvKEIB3UAEFBwgBQBKANyLFNeqADKPALbgAKnXAHQTHhqB0UQUG8ACwxQSEieLVRQbH8PAQBrcCTMSAhBBUAyAmVxSXVNbQBGAJNzS2oAHncvX3oAPlsHIA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQHYHsBaAQ2mmIE9CBjfXZAFzmof2nSnwDMvIAacNkgABBhQAOKatACW4hoRTwZuBgHoCJMpRp1GzVuwyREZNpEEBfEJaA

### Repro Code

```typescript
// This gets marked as error correct.
const result2 = new Array();

type SomeType = any;

// This should be marked as error…
const result1 = new Array<SomeType>();
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/no-array-constructor": ["error"],
  },
};
```

### Expected Result

Error for `new Array<SomeType>()`.

### Actual Result

No error for `new Array<SomeType>()`.

**Comment by bradzacher** [comment]
This is working as expected. See the valid examples in the rule's docs.
https://typescript-eslint.io/rules/no-array-constructor/

The reasoning is that a generic is safe to use because it explicitly types the array's elements.

If you just want to ban it altogether, then use the base rule rather than our extension.
